Client/src/Unified/Services/Submitter/BatchUntilInactiveBlock.cs

```diff
@@ -0,0 +1,249 @@
+    => source_.BatchSize;
+
+  /// <summary>
+  ///   Return the TimeSpan timer set in the constructor
+  /// </summary>
+  public TimeSpan Timeout { get; }
+
+  /// <summary>
+  ///   Task to check the completion
+  /// </summary>
+  public Task Completion
+    => source_.Completion;
+
+  /// <summary>
+  ///   Signal that the pipeline should be closed
+  /// </summary>
+  public void Complete()
+    => source_.Complete();
+
+  /// <summary>
+  ///   Return the exception produce in a Transform block or ActionBlock
+  /// </summary>
+  /// <param name="exception"></param>
+  public void Fault(Exception exception)
+    => ((IDataflowBlock)source_).Fault(exception);
+
+  /// <summary>
+  ///   Link block TransformBLock or ActionBlock
+  /// </summary>
+  /// <param name="target">the Block to link to</param>
+  /// <param name="linkOptions">
+  ///   Specific option to configure max message and propagation completion
+  ///   https://learn.microsoft.com/fr-fr/dotnet/api/system.threading.tasks.dataflow.dataflowlinkoptions?view=net-6.0
+  /// </param>
+  /// <returns></returns>
+  public IDisposable LinkTo(ITargetBlock<T[]>   target,
+                            DataflowLinkOptions linkOptions)
+    => timeoutTransformBlock_.LinkTo(target,
+                                     linkOptions);
+
+  /// <summary>
+  ///   To signal to the propagator object that a message is coming.
+  ///   Check if the pipeline is opened and the message is accepted
+  /// </summary>
+  /// <param name="messageHeader"></param>
+  /// <param name="messageValue"></param>
+  /// <param name="source"></param>
+  /// <param name="consumeToAccept"></param>
+  /// <returns></returns>
+  public DataflowMessageStatus OfferMessage(DataflowMessageHeader messageHeader,
+                                            T                     messageValue,
+                                            ISourceBlock<T>       source,
+                                            bool                  consumeToAccept)
+  {
+    var offerResult = ((ITargetBlock<T>)source_).OfferMessage(messageHeader,
+                                                              messageValue,
+                                                              source,
+                                                              consumeToAccept);
+
+    if (offerResult == DataflowMessageStatus.Accepted)
+    {
+      timer_.Change(Timeout,
+                    System.Threading.Timeout.InfiniteTimeSpan);
+    }
+
+    return offerResult;
+  }
+
+  /// <summary>
+  ///   Called by a linked ITargetBlock TInput  to accept and consume a
+  ///   DataFlowMessageHeader previously offered by this ISourceBlockTOutput .
+  /// </summary>
+  /// <param name="messageHeader"></param>
+  /// <param name="target"></param>
+  /// <param name="messageConsumed"></param>
+  /// <returns></returns>
+  public T[]? ConsumeMessage(DataflowMessageHeader messageHeader,
+                             ITargetBlock<T[]?>    target,
```

Comment:
Do you really need the array to be optional in `ITargetBlock` and return?
It is not the case for all the other methods.

Reply:
Nullable is not enabled anyway.
It may come from the interface.

Reply:
It was proposed by the interface. Project option will be set in https://github.com/aneoconsulting/ArmoniK/issues/647

Reply:
Checking the documentation (https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.dataflow.isourceblock-1.consumemessage?view=net-6.0), I can assure you that target type parameter should not be nullable (same in your cast). The output is, though.

```suggestion
                             ITargetBlock<T[]>     target,
```